src/App.tsx

```diff
@@ -1,26 +1,158 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterTypes';
+import { filterTodos } from './utils/helpers';
+import { ErrorTypes } from './types/ErrorTypes';
+import { Header } from './Components/Header';
+import { Footer } from './Components/Footer';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoList } from './Components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [error, setError] = useState<string>('');
+  const [value, setValue] = useState('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, tempTodo]);
+
+  const handleError = (errorMessage: ErrorTypes) => {
+    setError(errorMessage);
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .then(() => {
+        if (isLoading.length > 0) {
+          setIsLoading([]);
+        }
+      })
+      .catch(() => {
+        handleError(ErrorTypes.OneMessage);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const removeTodo = (todoId: number) => {
+    setIsLoading(prev => [...prev, todoId]);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+        setIsLoading([]);
+      })
+      .catch(() => {
+        setIsLoading([]);
+        handleError(ErrorTypes.UnableToDelete);
+      });
```

Comment:
```suggestion
        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
      })
      .catch(() => {
        handleError(ErrorTypes.UnableToDelete);
      })
      .finally(() => {
        setIsLoading(prevIds => prevIds.filter(id => id !== todoId));
      });
```
And remove loaders only what you actually need